packages/amplify-cli/src/extensions/amplify-helpers/push-resources.ts

```diff
@@ -49,6 +49,8 @@ export async function pushResources(
     }
   }
 
+  // building all CFN stacks here to get the resource Changes
+  context.amplify.executeProviderUtils(context, 'awscloudformation', 'buildOverrides', { forceCompile: true });
```

Comment:
please use constants here for awscloudformation and buildOverrides instead of strings